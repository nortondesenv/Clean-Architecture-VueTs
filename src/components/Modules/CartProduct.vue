<template>
    <div class="Cart__Product">
        <div class="Cart__Product__Content">
            <div style="Cart__Product__Content__Image">
                <img :src="item.image" />
            </div>
            <div class="Cart__Product__Content__Info">
                <div
                    class="Cart__Product__Content__Info__Name Cart__Product__Content__Info__Name--text-dark"
                >
                    {{ item.name }}
                </div>

                <div
                    @click="deleteItem(index)"
                    class="Cart__Product__Content__Info__Close Cart__Product__Content__Info__Close--text-light-dark"
                >
                    X
                </div>

                <div class="Cart__Product__Content__Info__Count">
                    <div class="Cart__Product__Content__Info__Count__Box">
                        <div
                            @click="lessOne()"
                            class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--icon"
                        >
                            -
                        </div>
                        <div
                            class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--text-medium"
                        >
                            {{ item.amount }}
                        </div>
                        <div
                            @click="addItem()"
                            class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--icon"
                        >
                            +
                        </div>
                    </div>
                </div>

                <div
                    class="Cart__Product__Content__Info__Price Cart__Product__Content__Info__Price--text-medium"
                >
                    {{ priceItem }}
                </div>
            </div>
        </div>
        <hr class="Cart__Product__Separator" />
    </div>
</template>

<script lang="ts">
import Vue from 'vue';
import RemoveItemUseCase, {
    IRemoveItemUseCase,
} from '@/containers/cart/RemoveItemUseCase';
import { IProduct } from '@/entities/Product';
import { CartEntity } from '@/entities/Cart';
import ErrorService from '@/services/ErrorService';

export default Vue.extend({
    props: {
        item: {
            type: Object as () => IProduct,
        },
        index: {
            type: Number,
        },
    },
    methods: {
        addItem(): void {
            this.item.amount = this.item.amount + 1;
        },
        lessOne(): void {
            if (this.item.amount == 1) return;
            this.item.amount = this.item.amount - 1;
        },
        async deleteItem(index) {
            const params: IRemoveItemUseCase = {
                cart: new CartEntity(),
                errorService: new ErrorService({
                    context: 'Remove produto do carrinho',
                }),
            };

            await new RemoveItemUseCase(params).execute(index);
        },
    },
    computed: {
        priceItem(): String {
            const priceItem = this.item.checkoutPrice * this.item.amount;

            return priceItem.toLocaleString('pt-br', {
                style: 'currency',
                currency: 'BRL',
            });
        },
    },
});
</script>
