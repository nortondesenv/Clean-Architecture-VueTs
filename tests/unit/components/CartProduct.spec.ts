import { shallowMount } from '@vue/test-utils';
import CartProduct from '@/components/modules/CartProduct.vue';

describe('Header.vue', () => {
    it('Renderiza lista de navegação do menu', () => {
        const product: any = {
            available: true,
            image: 'testeImage.png',
            name: 'testProduct',
            priceMember: 5.55,
            pricePromotional: 5.55,
            priceStock: 5.55,
            checkoutPrice: 5.55,
            amount: 55,
        };
        const wrapper = shallowMount(CartProduct, {
            propsData: { item: product },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
