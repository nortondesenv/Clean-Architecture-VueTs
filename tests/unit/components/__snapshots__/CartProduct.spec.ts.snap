// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartProduct.vue Renderiza lista de navegação do menu 1`] = `
<div class="Cart__Product">
  <div class="Cart__Product__Content">
    <div><img src="testeImage.png"></div>
    <div class="Cart__Product__Content__Info">
      <div class="Cart__Product__Content__Info__Name Cart__Product__Content__Info__Name--text-dark">
        testProduct
      </div>
      <div class="Cart__Product__Content__Info__Close Cart__Product__Content__Info__Close--text-light-dark">
        X
      </div>
      <div class="Cart__Product__Content__Info__Count">
        <div class="Cart__Product__Content__Info__Count__Box">
          <div class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--icon">
            -
          </div>
          <div class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--text-medium">
            55
          </div>
          <div class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--icon">
            +
          </div>
        </div>
      </div>
      <div class="Cart__Product__Content__Info__Price Cart__Product__Content__Info__Price--text-medium">
        R$&nbsp;305,25
      </div>
    </div>
  </div>
  <hr class="Cart__Product__Separator">
</div>
`;

exports[`Header.vue Renderiza lista de navegação do menu 1`] = `
<div class="Cart__Product">
  <div class="Cart__Product__Content">
    <div><img src="testeImage.png"></div>
    <div class="Cart__Product__Content__Info">
      <div class="Cart__Product__Content__Info__Name Cart__Product__Content__Info__Name--text-dark">
        testProduct
      </div>
      <div class="Cart__Product__Content__Info__Close Cart__Product__Content__Info__Close--text-light-dark">
        X
      </div>
      <div class="Cart__Product__Content__Info__Count">
        <div class="Cart__Product__Content__Info__Count__Box">
          <div class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--icon">
            -
          </div>
          <div class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--text-medium">
            55
          </div>
          <div class="Cart__Product__Content__Info__Count__Box__Item Cart__Product__Content__Info__Count__Box__Item--icon">
            +
          </div>
        </div>
      </div>
      <div class="Cart__Product__Content__Info__Price Cart__Product__Content__Info__Price--text-medium">
        R$&nbsp;305,25
      </div>
    </div>
  </div>
  <hr class="Cart__Product__Separator">
</div>
`;
